packages/api/src/routes/middlewares/rate-limiting.ts

```diff
@@ -0,0 +1,105 @@
+import { DynamoStore } from "@metriport/core/command/rate-limiting/ddb-store";
+import {
+  MetriportError,
+  RateLimitOperation,
+  RateLimitWindow,
+  defaultOperationLimits,
+  oneMinuteInMs,
+  rateLimitOperations,
+  rateLimitPartitionKey,
+  rateLimitWindows,
+  routeMapForError,
+} from "@metriport/shared";
+import { NextFunction, Request, Response } from "express";
+import { RateLimitRequestHandler, rateLimit } from "express-rate-limit";
+import { getDB } from "../../models/db";
+import { Config } from "../../shared/config";
+import { getCxIdOrFail } from "../util";
+
+const defaultWindow = oneMinuteInMs;
+
+let rateLimiter: RateLimitRequestHandler | undefined;
+export function initRateLimiter(): void {
+  rateLimiter = createRateLimiter();
+}
+
+/**
+ * Checks the CX request for the given operation and rate limit with fixed window.
+ */
+export function checkRateLimit(
+  operation: RateLimitOperation,
+  window: RateLimitWindow = defaultWindow
```

Comment:
I should've mentioned this earlier, but do you foresee that we need to have custom windows per routes?

It's nice to be extendable, but it adds a bit of complexity on the whole solution, whereas we would only need to check the operation if this was not an option.

Reply:
no longer relevant.